Simplify explanation of fleet-level user permissions (#43676)
fleetdm/fleet: articles/role-based-access.md

Users in Fleet either have global access or access to specific fleets.

> Users with access to specific fleets only have access to data (e.g. [hosts](https://fleetdm.com/docs/using-fleet/rest-api#hosts), [software](https://fleetdm.com/docs/using-fleet/rest-api#software), [policies](https://fleetdm.com/docs/using-fleet/rest-api#policies)) for their fleet.

Author: ```suggestion
Users with access to specific fleets only have access to data for their assigned fleet.
```

> Users with global access have access to data for all fleets. Check out [the user permissions

Author: ```suggestion
Global users can access data for all fleets. Check out [the user permissions
```

table](#user-permissions) above for global user permissions.

> Users can be assigned to multiple fleets, and can be assigned different roles for each fleet. For example, a user can be given access to the "💻 Workstations" fleet and assigned the "Observer" role. This same user can be given access to the "📱🔐 Personal mobile devices" fleet and assigned the "Maintainer" role.

Author: ```suggestion
Users can be assigned to multiple fleets, and can have different roles for each fleet. For example, a user can be given access to the "💻 Workstations" fleet and assigned the "Observer" role. This same user can be given access to the "📱🔐 Personal mobile devices" fleet and assigned the "Maintainer" role.
```
